src/Events/Services/EventsService.cs

```diff
@@ -64,31 +65,30 @@ public async Task<string> Save(CloudEvent cloudEvent)
         }
 
         /// <inheritdoc/>
-        public async Task<string> PostInbound(CloudEvent cloudEvent)
+        public async Task<string> PostRegistration(CloudEvent cloudEvent)
         {
-            QueuePostReceipt receipt = await _queueClient.EnqueueInbound(JsonSerializer.Serialize(cloudEvent));
+            cloudEvent.Id = Guid.NewGuid().ToString();
+            cloudEvent.Time = DateTime.UtcNow; // assign timestamp on initial receipt
```

Comment:
Setting time here without handling the db method that also sets time on the persisted cloud event will lead to inconsistent data

Reply:
Also wondering if controller should be responsible for assigning the guid and time, but can't quite make up my mind here. Argument being that the event service will then receive a valid cloudEvent instead of having to complete the object itself. On the other hand one could argue that the RequestModel is used as input to the service, and that the service is then responsible for both mapping and completing the object.

Reply:
I've added a new inserteventV2 function, need your help to manually test this.

Reply:
The new function has a different signature, so it doesn't need a new name as well. Pushed.